lib/praxis/plugins/praxis_mapper_plugin.rb

```diff
@@ -0,0 +1,203 @@
+require 'praxis-mapper'
+require 'singleton'
+
+require 'terminal-table'
+
+# Plugin for applications which use the 'praxis-mapper' gem.
+#
+# This plugin provides the following features:
+#   1. Sets up the PraxisMapper::IdentityMap for your application and assigns
+#      it to the Praxis::Application.instance.request.identity_map for access
+#      from your application.
+#   2. Connects to your database and dumps a log of database interaction stats
+#      (if enabled via the :log_stats option).
+#
+# This plugin accepts one of the following options:
+#   1. db_config: A Hash containing the configs for the database repositories
+#      queried through praxis-mapper. This parameter is a Hash where a key
+#      is the identifiers for a repository and the value is the options one
+#      would give to the 'sequel' gem. For example:
+#           db_config: {
+#             default: {
+#               host: 127.0.0.1,
+#               username: root,
+#               password: nil,
+#               database: myapp_dev,
+#               adapter: mysql2
+#             }
+#           }
+#   2. scope: A Hash containing scopes for the plugin to provide to the
+#      PraxisMapper::IdentityMap created.
+#   3. log_stats: A Boolean flag indicating whether or not to dump DB stats to
+#      the Praxis::Application.instance.logger app log.
+#   4. config_file: A String indicating the path where this plugin's config
+#      file exists.
+#
+# See http://praxis-framework.io/reference/plugins/ for further details on how
+# to use a plugin.
+#
+module Praxis
+  module Plugins
+    module PraxisMapperPlugin
+      include Praxis::PluginConcern
+
+      class RepositoryConfig < Attributor::Hash
+        self.key_type = String
+
+        keys allow_extra: true do
+          key 'type', String, default: 'sequel'
+          extra 'connection_settings'
+        end
+
+      end
+
+
+      class Plugin < Praxis::Plugin
+        include Singleton
+
+        def initialize
+          @options = {
+            config_file: 'config/praxis_mapper.yml',
+            scopes: {},
```

Comment:
Currently, the apps I'm thinking will use this add a scope, but they cannot do it at plugin initialization time, because the scope depends on the account_id in the global_session. I'm fine with removing this, though.